src/ImageSharp/Diagnostics/MemoryDiagnostics.cs

```diff
@@ -56,8 +63,16 @@ public static class MemoryDiagnostics
 
         internal static void RaiseUndisposedMemoryResource(string allocationStackTrace)
         {
+            if (undisposedAllocation is null)
+            {
+                return;
+            }
+
             // Schedule on the ThreadPool, to avoid user callback messing up the finalizer thread.
-            ThreadPool.QueueUserWorkItem(_ => undisposedMemoryResource?.Invoke(allocationStackTrace));
+            ThreadPool.QueueUserWorkItem(
+                stackTrace => undisposedAllocation?.Invoke(stackTrace),
+                allocationStackTrace,
+                preferLocal: true);
```

Comment:
```suggestion
                preferLocal: false);
```
I believe it should go to the global thread-pool queue not the local one.